identity_jose/src/jws/algorithm.rs

```diff
@@ -109,13 +141,16 @@ impl FromStr for JwsAlgorithm {
       "ES256K" => Ok(Self::ES256K),
       "none" => Ok(Self::NONE),
       "EdDSA" => Ok(Self::EdDSA),
-      _ => Err(Error::JwsAlgorithmParsingError),
+      #[cfg(feature = "custom_alg")]
+      value => Ok(Self::Custom(value.to_string())),
+      #[cfg(not(feature = "custom_alg"))]
+      _ => Err(crate::error::Error::JwsAlgorithmParsingError),
     }
   }
 }
 
 impl Display for JwsAlgorithm {
   fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-    f.write_str(self.name())
+    f.write_str(&self.clone().name())
```

Comment:
Hm, we still need to pass a `&str` to `write_str`, which would add a `&` to the version without clone:

```rs
    f.write_str(&(*self).name())
```

This would then produce a clippy warning for non `custom_alg` builds (as we'd then be passing `&&str` to `write_str`), asking us to change it back to:


```rs
f.write_str((*self).name())
```

I'd suggest to just offer 2 versions of the `Display` impl here as the result of `.name()` differs? Basically using the original version for non-`custom_alg` and the suggestion from above plus the `&` for `custom_alg`.

```rs
#[cfg(not(feature = "custom_alg"))]
impl Display for JwsAlgorithm {
  fn fmt(&self, f: &mut Formatter<'_>) -> Result {
    f.write_str(self.name())
  }
}

#[cfg(feature = "custom_alg")]
impl Display for JwsAlgorithm {
  fn fmt(&self, f: &mut Formatter<'_>) -> Result {
    f.write_str(&(*self).name())
  }
}
```

Reply:
Thanks @itsyaasir and @wulfraem. I split the implementation as you suggested.